django_ipam/templates/admin/django-ipam/subnet/change_form.html

```diff
@@ -70,6 +54,108 @@ <h3>{% trans 'Used IP address' %}</h3>
+      var lastRenderedPage = 0; //1 based indexing (0 -> no page rendered)
+      function pageContainer(page){
+        div = $('<div></div>');
+        for (const address of page) {
+          div.append(addressListItem(address));
+        }
+        return div;
+      }
+      function addressListItem(addr) {
+        if (addr.used) {
+          return '<li>\
+            <a class="used">\
+              '+ addr.address + '\
+            </a>\
+          </li>';
+        }
+        else {
+          return '<li>\
+            <a href=\"{% url ipaddress_add_url %}?_to_field=id&amp;_popup=1&amp;ip_address='+ addr.address + '&amp;subnet=' + current_subnet + '\"\
+               onclick=\"return showAddAnotherPopup(this);\">\
+              '+ addr.address + '\
+            </a>\
+          </li>'
+        }
+      }
+      function pageDown() {
+        busy = true;
+        if (fetchedPages.length > lastRenderedPage) {
+          appendPage();
+        }
+        else if (nextPageUrl != null) {
+          fetchNextPage();
+        }
+        else {
+          busy = false;
+        }
+      }
+      function pageUp() {
+        busy = true;
+        if (lastRenderedPage > renderedPages) {
+          $('.subnet-visual div:last').remove();
+          addedDiv = pageContainer(fetchedPages[lastRenderedPage - renderedPages - 1]);
+          $('.subnet-visual').prepend(addedDiv);
+          $('.subnet-visual').scrollTop($('.subnet-visual').scrollTop()+addedDiv.height());
+          lastRenderedPage--;
+        }
+        busy = false;
+      }
+      function appendPage() {
+        $('.subnet-visual').append(pageContainer(fetchedPages[lastRenderedPage]));
+        if (lastRenderedPage >= renderedPages) {
+          removedDiv = $('.subnet-visual div:first');
+          $('.subnet-visual').scrollTop($('.subnet-visual').scrollTop()-removedDiv.height());
+          removedDiv.remove();
+        }
+        lastRenderedPage++;
+        busy = false;
+        onUpdate();
+      }
+      function fetchNextPage() {
+        $.ajax({
+          type: 'GET',
+          url: nextPageUrl,
+          success: function (res) {
+            fetchedPages.push(res.results)
+            nextPageUrl = res.next;
+            appendPage();
+          },
+          error: function (error) {
+            busy = false;
+            throw error;
+          },
+        });
+      }
+      function onUpdate() {
+        if (!busy) {
+          const scrollTop = $('.subnet-visual').scrollTop();
+          const scrollBottom = scrollTop + $('.subnet-visual').innerHeight();
+          const height = $('.subnet-visual')[0].scrollHeight;
+          if (height * 0.75 <= scrollBottom) {
+            pageDown();
+          }
+          else if (height * 0.25 >= scrollTop) {
+            pageUp();
+          }
+        }
+      }
+      $('.subnet-visual').scroll(onUpdate);
+      onUpdate();
+    }(django.jQuery));
```

Comment:
try to turn this into a function loaded from an external file, make sure the function accepts parameters because we'll need to pass values like `{{ subnet.id }}`